mr/src/test/java/org/apache/iceberg/mr/TestIcebergInputFormats.java

```diff
@@ -90,52 +93,49 @@ public class TestIcebergInputFormats {
   private static final PartitionSpec SPEC =
       PartitionSpec.builderFor(SCHEMA).identity("date").bucket("id", 1).build();
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir public Path temp;
 
   // before variables
   private Configuration conf;
   private TestHelper helper;
   private InputFormatConfig.ConfigBuilder builder;
 
   // parametrized variables
-  private final TestInputFormat.Factory<Record> testInputFormat;
-  private final FileFormat fileFormat;
+  @Parameter(index = 0)
+  private TestInputFormat.Factory<Record> testInputFormat;
+
+  @Parameter(index = 1)
+  private FileFormat fileFormat;
 
-  @Before
+  @BeforeEach
   public void before() throws IOException {
     conf = new Configuration();
     conf.set(CatalogUtil.ICEBERG_CATALOG_TYPE, Catalogs.LOCATION);
     HadoopTables tables = new HadoopTables(conf);
 
-    File location = temp.newFolder(testInputFormat.name(), fileFormat.name());
-    Assert.assertTrue(location.delete());
+    File location = temp.resolve(Paths.get(testInputFormat.name(), fileFormat.name())).toFile();
```

Comment:
in other places we use `Files.createTempDirectory(temp, "junit").toFile()`, so I think we should do the same here

Reply:
In other places, the  `Files.createTempDirectory(temp, "junit").toFile()` is used to replace `temp.newFolder();` which create a `junit` folder under the `temp`'s root path.
However, when comes `temp.newFolder(path1, path2)`, the method create a new leveled path like `${tempRoot}/path1/path2`. I read the source code of `Files.createTempDirectory` and it can only create one level path. So I use `temp.resolve` here